Authentication/phpBBSessionAuthenticator.php

```diff
@@ -31,27 +35,61 @@ class phpBBSessionAuthenticator implements SimplePreAuthenticatorInterface, Auth
     /** @var  string */
     private $loginpage;
 
+    /** @var RequestStack  */
+    private $requestStack;
+
+    /** @var ContainerInterface  */
+    private $container;
+
+    /** @var  string */
+    private $dbconnection;
+
     /**
      * @param $cookiename string
      * @param $boardurl  string
      * @param $loginpage string
+     * @param $requestStack RequestStack
+     * @param ContainerInterface $container
      */
-    public function __construct($cookiename, $boardurl, $loginpage)
+    public function __construct($cookiename, $boardurl, $loginpage, $dbconnection, RequestStack $requestStack, ContainerInterface $container)
     {
-        $this->cookiename = $cookiename;
-        $this->boardurl   = $boardurl;
-        $this->loginpage  = $loginpage;
-
+        $this->cookiename   = $cookiename;
+        $this->boardurl     = $boardurl;
+        $this->loginpage    = $loginpage;
+        $this->dbconnection = $dbconnection;
+        $this->requestStack = $requestStack;
+        $this->container    = $container;
     }
 
     /**
      * @param TokenInterface $token
      * @param UserProviderInterface $userProvider
      * @param $providerKey
+     * @return AnonymousToken
      */
     public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
     {
-        // TODO: Implement authenticateToken() method.
+        if (!$userProvider instanceof phpBBUserProvider)
+        {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'The user provider must be an instance of phpBBUserProvider (%s was given).',
+                    get_class($userProvider)
+                )
+            );
+        }
+
+        $session_id = $this->requestStack->getCurrentRequest()->cookies->get($this->cookiename . '_sid');
+
+        if (empty($session_id))
+        {
+            return null; // We can't authenticate if no SID is available.
```

Comment:
@Nicofuma We might support it, however is there a chance that there is actually a SID in the URL in a symfony page? 
I suppose symfony won't keep it after the first page.